### DISCUSSION TITLE
Aspect ratio issues

### DISCUSSION THREAD

**Discussion by tyfoon** [blakeblackshear/frigate#4220] [General]
For some reason my Hikvision cams allow me to specify a 4:3 (640x480) aspect ratio in the sub stream, this is also the highest res possible for the substream. And this while the sensor in the camera is 16:9 so the image looks 'squeezed' in Frigate. This substream is used for detection only.  

My mainstream -which is setup in 16:9 (2688/1520 resolution) in the camera- is defined as RTMP & Record in frigate config. 

As I use mostly HASS with frigate card I don't mind that much that the video in frigate webview is squeezed into the 4:3 aspect ratio. However HASS (via the Frigate card) also takes over that behavior. I'm a bit puzzled by this as I would expect HASS frigate card to take the RTMP defined high res stream which is 16:9 (set via de Cam). The frigate card has some options to change aspect ratio. They don't really help as the image stays either in 4:3 or a 4:3 is in a 16:9 window (creating black bars left & right).

Any suggestions where to fix this behavior welcome!

**Comment by NickM-27** [comment]
You should be setting the frigate `detect -> width / height` according to the actual resolution of the detect stream. As far as the frigate card goes, depends which stream source you are selecting when setting the card up

**Reply by tyfoon** [reply]
> You should be setting the frigate `detect -> width / height` 

This is set identical to the stream. I think frigate actually would not work according to the docs. So frigate correctly reads the 'wrong' aspect ration from the camera (detect stream). My question was about the RTMP stream which is 16:9.

However update for anyone that might also run into this issue:
I just installed the [webrtc addon](https://community.home-assistant.io/t/add-on-rtsptoweb-and-rtsptowebrtc/387846) and this solved it without changing anything else (also not in the frigate card). It seems that if you do not have webrtc installed the frigate card takes the detect stream (or at least the detect aspect ratio). When it finds the webrtc install it seems to pick the rtmp stream correctly.
